terraform/shared/scripts/create_elastic_users_identity.py

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env python
+"""
+This script is run by manually after to create roles/users in
+an Elastic Cloud cluster immediately after it's been created.
+"""
+
+import functools
+import secrets
+
+import boto3
+import click
+
+from elasticsearch import Elasticsearch
+
+SERVICES = {
+    "requests": ["catalogue_read"],
+}
+
+
+@functools.lru_cache()
+def get_aws_client(resource, *, role_arn):
+    """
+    Get a boto3 client authenticated against the given role.
+    """
+    sts_client = boto3.client("sts")
+    assumed_role_object = sts_client.assume_role(
+        RoleArn=role_arn, RoleSessionName="AssumeRoleSession1"
+    )
+
+    credentials = assumed_role_object["Credentials"]
+
+    return boto3.client(
+        resource,
+        aws_access_key_id=credentials["AccessKeyId"],
+        aws_secret_access_key=credentials["SecretAccessKey"],
+        aws_session_token=credentials["SessionToken"],
+    )
+
+
+def read_secret(secret_id, role_arn):
+    """
+    Retrieve a secret from Secrets Manager.
+    """
+    secrets_client = get_aws_client("secretsmanager", role_arn=role_arn)
+
+    return secrets_client.get_secret_value(SecretId=secret_id)["SecretString"]
+
+
+def store_secret(secret_id, secret_value, role_arn):
+    """
+    Store a key/value pair in Secrets Manager.
+    """
+
+    secrets_client = get_aws_client("secretsmanager", role_arn=role_arn)
+
+    resp = secrets_client.put_secret_value(
+        SecretId=secret_id, SecretString=secret_value
+    )
+
+    if resp["ResponseMetadata"]["HTTPStatusCode"] != 200:
+        raise RuntimeError(f"Unexpected error from PutSecretValue: {resp}")
+
+    click.echo(f"Stored secret {click.style(secret_id, 'yellow')}")
+
+
+if __name__ == '__main__':
+    catalogue_role_arn = "arn:aws:iam::756629837203:role/catalogue-developer"
+    catalogue_secret_prefix = f"elasticsearch/catalogue_api"
+
+    username = read_secret(f"{catalogue_secret_prefix}/username", catalogue_role_arn)
+    password = read_secret(f"{catalogue_secret_prefix}/password", catalogue_role_arn)
+
+    role_arn = "arn:aws:iam::770700576653:role/identity-developer"
+    secret_prefix = f"elasticsearch/catalogue_api"
```

Comment:
```suggestion
    identity_role_arn = "arn:aws:iam::770700576653:role/identity-developer"
    identity_secret_prefix = f"elasticsearch/catalogue_api"
```

to distinguish from the catalogue variables.

Reply:
Can't commit this as used later on, will refactor locally.